lib/collectors.js

```diff
@@ -2,6 +2,46 @@
 
 module.exports = (api) => {
 
+  api.metasync.collect = (expected) => {
+
+    const isCount = typeof(expected) === 'number';
+    const isKeys = Array.isArray(expected);
+    if (!(isCount || isKeys)) throw new Error('Unexpected type');
+    const keys = expected;
+    if (isKeys) expected = expected.length;
+    let count = 0;
+    let timeout = 0;
+    let timer = null;
+    let done = null;
+    const data = {};
+
+    const collector = (key, value) => {
+      if (isCount || keys.includes(key)) count++;
+      if (value instanceof Error) return done(value, data);
+      data[key] = value;
+      if (expected === count) {
+        if (timer) clearTimeout(timer);
+        done(null, data);
+      }
+      return collector;
+    };
+
+    collector.timeout = (milliseconds) => {
+      timeout = milliseconds;
```

Comment:
Do we really need to store the timeout value itself? I'd say we can get rid of that variable and only store the timer object in the closure.